packages/terra-collapsible-menu-view/src/CollapsibleMenuView.jsx

```diff
@@ -107,7 +107,7 @@ class CollapsibleMenuView extends React.Component {
 
       if (calcWidth > availableWidth) {
         // If last child fits in the available space, leave it face up
-        if (i === this.props.children.length - 1 && calcWidth <= width) {
```

Comment:
I vaguely remember this `-1` being pretty important. Has this been tested in all scenarios? Specifically when the there are multiple items without showing the collapsible icon? 

I believe this check was for when the last button was roughly the same size as the ellipsis button. We should show the actual button instead of the ellipsis if it can fit.

Reply:
I think you're right, but It looks like we should find another way to accomplish this, as I'm not sure this is even fully fixing the original intent. Basically we're just exempting the last item in a list from length checking regardless of whether it fits or not. This won't affect single children only when the child is a single item in a list.

Reply:
We should probably investigate this css rule:
https://github.com/cerner/terra-framework/blob/master/packages/terra-collapsible-menu-view/src/CollapsibleMenuView.module.scss#L36
Should we be constraining the width of our items/groups to the width of the menu view or not, that might affect the last item logic as well.

Reply:
So... removing the `-1` means that this `if` check never evaluates to `true`. The  index of the last item in the array is the array's `length`  - 1.

Reply:
The problem here is really that when react has one child, children is an element and not an array. So this probably just needs to be updated to
`if (i === React.Children.count(this.props.children) - 1 && calcWidth <= width) {`

Reply:
@StephenEsser, yes removing `-1` does create a problem with the ellipsis. So, now I have found that the bug is because of the function `getBoundingClientRect().width` as it only provides the size of the child in the current window size (length of visible window) and not for the overflown elements. I tried using `scrollWidth, offsetWidth, and getComputedStyle` properties but it didn't work out. I was getting the required result with `scrollWidth` but it was breaking for IE11 because of a bug in it but seemed to work fine with IE10.
 @jmsv6d, I tried this code `if (i === React.Children.count(this.props.children) - 1 && calcWidth <= width)`  but it didn't solve the issue. 
**I tried implementing it like this -:**
``` JAVASCRIPT
  handleResize(width) {
    const menuButtonWidth = this.menuButton.getBoundingClientRect().width;
    const availableWidth = width - menuButtonWidth;
    let hiddenStartIndex = -1;
    let calcWidth = 0;
    let menuHidden = true;

    for (let i = 0; i < React.Children.count(this.props.children); i += 1) {
      const child = this.container.children[i];
      if (this.props.children.length) === 1) {
        for (let j = 0; j < child.children.length; j += 1) {
          const childWidth = child.children[j].getBoundingClientRect().width;
          calcWidth += childWidth;
        }
      } else {
        const childWidth = child.getBoundingClientRect().width;
        calcWidth += childWidth;
      }
 
      if (calcWidth > availableWidth) {
        // If last child fits in the available space, leave it face up
        if (i === this.props.children.length - 1 && calcWidth <= width) {
          break;
        }

        hiddenStartIndex = i;
        menuHidden = false;
        break;
      }
    }
```

In this logic, when there is only 1 child with multiple elements or even 1 element it checks for the childwidth individually. I tried  
``` JAVASCRIPT
for (let j = 0; j < child.children.length; j += 1) {
          const childWidth = child.children[j].getBoundingClientRect().width;
          calcWidth += childWidth;
```
implementing it without the if-else condition will cause an error of width being undefined.

Reply:
@pranav300 Have you tried removing the style @dkasper-was-taken mentioned to see how that effects the width calculation?

Reply:
@jmsv6d I didn't remove the style but changed its value and it didn't seem to work at that time, But ya removing the style does solve the issue and doesn't affect the last item logic.